### ISSUE TITLE
pgstat_send_qd_tabstats can not work in REFRESH MATERIALIZED VIEW 

### ISSUE THREAD

**Issue by hidva** [greenplum-db/gpdb#11375]
### Greenplum version or build

0a4101492a4196a713f3d0d5fe687d5b9e0fc3ef

### Expected behavior

```
tmp=# create table t(i int, j int, z int );
CREATE TABLE
tmp=# insert into t select i,i,i from generate_series(1, 100) i;
INSERT 0 100
tmp=#  create materialized view mt as select * from t where z>=50;
SELECT 51
tmp=# select n_tup_ins, n_tup_upd, n_tup_del, n_tup_hot_upd, n_live_tup, n_dead_tup, n_mod_since_analyze from pg_stat_all_tables where relid = 'mt'::regclass;
 n_tup_ins | n_tup_upd | n_tup_del | n_tup_hot_upd | n_live_tup | n_dead_tup | n_mod_since_analyze
-----------+-----------+-----------+---------------+------------+------------+---------------------
        51 |         0 |         0 |             0 |         51 |          0 |                  51
tmp=# insert into t select i,i,i from generate_series(1, 100) i;
INSERT 0 100
tmp=# refresh materialized view mt;
REFRESH MATERIALIZED VIEW
tmp=# select n_tup_ins, n_tup_upd, n_tup_del, n_tup_hot_upd, n_live_tup, n_dead_tup, n_mod_since_analyze from pg_stat_all_tables where relid = 'mt'::regclass;
 n_tup_ins | n_tup_upd | n_tup_del | n_tup_hot_upd | n_live_tup | n_dead_tup | n_mod_since_analyze
-----------+-----------+-----------+---------------+------------+------------+---------------------
       153 |         0 |         0 |             0 |        102 |          0 |                 153
```

### Actual behavior

```
tmp=# create table t(i int, j int, z int );
CREATE TABLE
tmp=# insert into t select i,i,i from generate_series(1, 100) i;
INSERT 0 100
tmp=#  create materialized view mt as select * from t where z>=50;
SELECT 51
tmp=# select n_tup_ins, n_tup_upd, n_tup_del, n_tup_hot_upd, n_live_tup, n_dead_tup, n_mod_since_analyze from pg_stat_all_tables where relid = 'mt'::regclass;
 n_tup_ins | n_tup_upd | n_tup_del | n_tup_hot_upd | n_live_tup | n_dead_tup | n_mod_since_analyze
-----------+-----------+-----------+---------------+------------+------------+---------------------
        51 |         0 |         0 |             0 |         51 |          0 |                  51
tmp=# insert into t select i,i,i from generate_series(1, 100) i;
INSERT 0 100
tmp=# refresh materialized view mt;
REFRESH MATERIALIZED VIEW
tmp=# select n_tup_ins, n_tup_upd, n_tup_del, n_tup_hot_upd, n_live_tup, n_dead_tup, n_mod_since_analyze from pg_stat_all_tables where relid = 'mt'::regclass;
 n_tup_ins | n_tup_upd | n_tup_del | n_tup_hot_upd | n_live_tup | n_dead_tup | n_mod_since_analyze
-----------+-----------+-----------+---------------+------------+------------+---------------------
        51 |         0 |         0 |             0 |          0 |          0 |                   0
```

**Comment by hidva** [comment]
@JunfengYang

**Comment by JunfengYang** [comment]
Ahhh, just find the root cause. This is caused by a small bug.
https://github.com/greenplum-db/gpdb/blob/8da60594c0bb4eddfaea5d37670a3f0efe07f03e/src/backend/commands/matview.c#L457-L467

This line `processed = queryDesc->estate->es_processed; ` here we haven't collected the total processed tuples yet.
`ExecutorEnd` is responsible for collecting processed tuples.
We should get it after `ExecutorEnd`.

**Comment by asimrp** [comment]
@JunfengYang you are right, that `es_processed` is updated in `ExecutorEnd()` but why does it work in upstream PostgreSQL?  This part of the code is inherited as-is from upstream.  Also, we seem to have merged this commit from upstream: 
```
commit 17f8ffa1e331cd0d95a3c4ccec66ea83d8b893c5
Author: Tom Lane <tgl@sss.pgh.pa.us>
Date:   Sat Mar 18 17:49:06 2017 -0400

    Fix REFRESH MATERIALIZED VIEW to report activity to the stats collector.
```

Is something else going wrong here?

**Comment by JunfengYang** [comment]
@asimrp We do the dispatch here and the `es_processed` is 0 before we fetch the total processed across segments.
And pg don't do dispatch, it keeps increase `es_processed`.

**Comment by JunfengYang** [comment]
Merged the fix PR as b1d8049258f450ae51b93a9072bd974fd8f68079 in Master branch. So close it.
